code/modules/admin/topic.dm

```diff
@@ -2868,9 +2868,12 @@
 				SSblackbox.record_feedback("tally", "admin_secrets_fun_used", 1, "Traitor All ([objective])")
 
 				for(var/mob/living/carbon/human/H in GLOB.player_list)
-					if(H.stat == 2 || !H.client || !H.mind) continue
-					if(is_special_character(H)) continue
-					//traitorize(H, objective, 0)
+					if(H.stat == 2 || !H.client || !H.mind)
+						continue
+					if(is_special_character(H))
+						continue
+					if(jobban_isbanned(H, ROLE_TRAITOR) || jobban_isbanned(H, "Syndicate"))
```

Comment:
I was gonna suggest to use `CandCheck` here. But that one is a bit bloated/too specific for the purpose here.

Reply:
I personally have no idea what that is so I'll take your word for it either way :p